src/GraphQLParser/ROM.cs

```diff
@@ -40,39 +40,46 @@ public ROM(ReadOnlyMemory<char> memory)
     /// <inheritdoc/>
     public bool Equals(ROM other)
     {
-        if (_memory.Length != other._memory.Length)
-            return false;
-
+        // fast check in case of memory is backed by the same string object
+        //public bool Equals(ReadOnlyMemory<T> other)
+        //{
+        //    if (_object == other._object && _index == other._index)
+        //    {
+        //        return _length == other._length;
+        //    }
+
+        //    return false;
+        //}
+        if (_memory.Equals(other._memory))
+            return true;
+
+        // then check byte by byte, SequenceEqual already has length check
+        //[MethodImpl(MethodImplOptions.AggressiveInlining)]
+        //public static bool SequenceEqual<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other) where T : IEquatable<T>
+        //{
+        //    int length = span.Length;
+        //    if (default(T) != null && IsTypeComparableAsBytes<T>(out NUInt size))
+        //    {
+        //        if (length == other.Length)
+        //        {
+        //            return SpanHelpers.SequenceEqual(ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)), ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(other)), (NUInt)length * size);
+        //        }
+
+        //        return false;
+        //    }
+
+        //    if (length == other.Length)
+        //    {
+        //        return SpanHelpers.SequenceEqual(ref MemoryMarshal.GetReference(span), ref MemoryMarshal.GetReference(other), length);
+        //    }
+
+        //    return false;
+        //}
         return _memory.Span.SequenceEqual(other._memory.Span);
```

Comment:
Might want to check if this method is faster for .NET Standard 2.1:

https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.equals?view=net-6.0#system-memoryextensions-equals(system-readonlyspan((system-char))-system-readonlyspan((system-char))-system-stringcomparison)

Reply:
I did not check. It has already been used. I only added an additional call above which should work pretty quickly.

Reply:
Well if you are planning to use this call extensively, you might want to.

Reply:
Oh, it's another method. OK, added TODO to check later.